llvm/lib/IR/RepoDefinition.cpp

```diff
@@ -90,24 +90,35 @@ const Constant *getAliasee(const GlobalAlias *GA) {
 
 template <typename GlobalType>
 void ModuleHashGenerator::calculateGOInfo(const GlobalType *G) {
-  GOInfo Result = calculateDigestAndDependenciesAndContributedToGVs(G);
-  // ContributedToGVs of an object is used to update other objects'
-  // contributions. For example, if the ContributedToGVs of function `foo`  are
-  // global variables `g` and `q`, the function `foo` is in the contributions of
-  // `g` and `q`.
-  // ContributedToGVs[`foo`] = [`g`, `q`]
-  // ====> Contributions[`g`] = [`foo`],
-  //       Contributions[`q`] = [`foo`]
+  GODetailedInfo Result = calculateDigestAndDependenciesAndContributions(G);
+#ifndef NDEBUG
+  LLVM_DEBUG(dbgs() << "\nGO Name:" << G->getName() << "\n");
+  LLVM_DEBUG(Result.dump());
+#endif
+  // Update G's dependencies.
```

Comment:
If I’m reading it correctly, the code copies the dependencies from `HashInfo` (could it move instead?) then converts contributions into dependency records. I’m not sure that this comment really reflects that!

Reply:
Updated the comment in https://github.com/SNSystems/llvm-project-prepo/pull/127/commits/4a8756041970f24b02df38128f15c683126a7ebe.